File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
   
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,bool>mp;
        int j = 0, n = s.size(),ans = 0;
        for(int i = 0 ; i < s.size();i++){
            while(j < n && mp[s[j]] == false){
                mp[s[j]]  = true;
                j+=1;
                ans=max(ans,j-i);
            }
            mp[s[i]] = false;
        }
        return ans;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = mx = 0
        vis = set()
        for r in range(len(s)):
            while s[r] in vis:
                vis.remove(s[l])
                l+=1
            
            vis.add(s[r])
            mx=max(mx,r-l+1)
        return mx
            

